portal/static/js/main.js

```diff
@@ -3061,101 +3125,100 @@ var Profile = function(subjectId, currentUserId) {
-                                });
-                            });
-                            if ($("input[name='pca_diag']").length > 0) {
-                                tnthAjax.putClinical(self.subjectId,"pca_diag","false", $(this));
-                            };
-                            if ($("input[name='pca_localized']").length > 0) {
-                                tnthAjax.putClinical(self.subjectId,"pca_localized","false", $(this));
-                            };
-                        } else if (toCall == "pca_diag") {
-                            ["pca_localized"].forEach(function(fieldName) {
-                              $("input[name='" + fieldName + "']").each(function() {
-                                  $(this).prop("checked", false);
-                              });
-                            });
-                            if ($("input[name='pca_localized']").length > 0) {
-                                tnthAjax.putClinical(self.subjectId,"pca_localized","false", $(this));
-                            };
+        if (hasValue(self.subjectId)) {
+            $(".pat-q input:radio").on("click",function(){
+                var thisItem = $(this);
+                var toCall = thisItem.attr("name")
+                // Get value from div - either true or false
+                var toSend = thisItem.val();
+                if (String(toCall) !== "biopsy") {
+                    tnthAjax.postClinical(self.subjectId,toCall,toSend, $(this).attr("data-status"), $(this));
+                }
+                if (String(toSend) === "true" || String(toCall) ===  "pca_localized") {
+                    if (String(toCall) === "biopsy") {
+                        if ($("#biopsyDate").val() === "") {
+                            return true;
+                        }
+                        else {
+                          tnthAjax.postClinical(self.subjectId, toCall, toSend, "", $(this), {"issuedDate": $("#biopsyDate").val()});
                         }
-                        thisItem.parents(".pat-q").nextAll().fadeOut();
                     };
-                });
-            };
+                    thisItem.parents(".pat-q").nextAll().fadeIn();
+                } else {
+                    if (String(toCall) === "biopsy") {
+                        tnthAjax.postClinical(self.subjectId, toCall, "false", $(this).attr("data-status"), $(this));
+                        ["pca_diag", "pca_localized"].forEach(function(fieldName) {
+                            $("input[name='" + fieldName + "']").each(function() {
+                                $(this).prop("checked", false);
+                            });
+                        });
+                        if ($("input[name='pca_diag']").length > 0) {
+                            tnthAjax.putClinical(self.subjectId,"pca_diag","false", $(this));
+                        };
+                        if ($("input[name='pca_localized']").length > 0) {
```

Comment:
I *think* this `if` block and its matching one in `else if (String(toCall) === "pca_diag")` could be moved to the `else` block that is containing them both (removing one copy).

Reply:
@ivan-c Ivan, I think it was purposefully coded that way because we also needed to consider start treatment question - in which case we won't be setting the localized (i.e. pca_diag) variable at all.